api/census.go

```diff
@@ -199,10 +199,26 @@ func (a *API) addCensusParticipantsHandler(w http.ResponseWriter, r *http.Reques
 
 	// if async create a new job identifier
 	jobID := internal.HexBytes(util.RandomBytes(16))
+
+	// Create persistent job record
+	if err := a.db.CreateJob(jobID.String(), db.JobTypeCensusParticipants, census.OrgAddress, len(members.Members)); err != nil {
+		log.Warnw("failed to create persistent job record", "error", err, "jobId", jobID.String())
+		// Continue with in-memory only (fallback)
+	}
+
 	go func() {
 		for p := range progressChan {
 			// We need to drain the channel to avoid blocking
 			addParticipantsToCensusWorkers.Store(jobID.String(), p)
+
+			// When job completes, persist final results
+			if p.Progress == 100 {
+				// BulkCensusParticipantStatus doesn't have errors, so pass empty slice
+				if err := a.db.CompleteJob(jobID.String(), p.Added, []string{}); err != nil {
```

Comment:
@altergui is it possible here to have the job completed but with errors and miss the errors?